integration_tests/src/main/python/fastparquet_compatibility_test.py

```diff
@@ -76,25 +76,38 @@ def read_with_fastparquet_or_plugin(spark):
                               start=datetime(1, 1, 1, tzinfo=timezone.utc),
                               end=datetime(1899, 12, 31, tzinfo=timezone.utc)),
                  marks=pytest.mark.xfail(reason="fastparquet reads timestamps preceding 1900 incorrectly.")),
-    #  TODO: Array gen type deduction borked when converting from Pandas to Spark dataframe.
-    # ArrayGen(child_gen=IntegerGen(nullable=False), nullable=False),
-    #  TODO: Struct rows seem to be correct, but are failing comparison because of differences in Row representation.
-    # StructGen(children=[("first", IntegerGen(nullable=False)),
-    #                     ("second", FloatGen(nullable=False))], nullable=False)
+    pytest.param(
+        ArrayGen(child_gen=IntegerGen(nullable=False), nullable=False),
+        marks=pytest.mark.xfail(reason="Conversion from Pandas dataframe to Spark dataframe fails: "
+                                       "\"Unable to infer the type of the field a\".")),
+    pytest.param(
+        StructGen(children=[("first", IntegerGen(nullable=False)),
+                            ("second", FloatGen(nullable=False))], nullable=False),
+        marks=pytest.mark.xfail(reason="Values are correct, but struct row representations differ between "
```

Comment:
Are there details in how they differ. Should we have a follow on issue to understand this better so we can file/fix any issues.

Reply:
This should be a `TODO:`, for follow-up. I will document it in the test.

The problem with the structs test is that the diff turns out as follows:
```
--- CPU OUTPUT
+++ GPU OUTPUT
@@ -1 +1 @@
-Row(a.first=-341142443, a.second=3.333994866005594e-37)
+Row(a=Row(first=-341142443, second=3.333994866005594e-37))
```
The difference looks notational to me.

Reply:
#9399